aten/src/ATen/core/type.cpp

```diff
@@ -75,12 +76,24 @@ AnyTypePtr AnyType::get() {
   return value;
 }
 
-template <typename T>
-static bool compatible_optional(c10::optional<T> e, T a) {
+template <typename T, typename Y = T>
+bool compatible_optional(c10::optional<T> e, Y a) {
   return !e.has_value() || e.value() == a;
 }
 
-static bool compatible_varying_shape(const VaryingShape& e, at::IntArrayRef a) {
+template <typename T, int64_t>
```

Comment:
This is a lot of extra code to avoid creating a SymbolicShape object from an integer. I would prefer explicitly doing that over this complicated template logic.